### PR TITLE
[30] Prevent surveys with duplicate names from being added.

### LINKED ISSUE #30: Adding duplicate surveys deletes that survey
Steps for reproduction:

1. Create a survey with any name
2. Create another survey with the same name

Expected result: Should either throw an error or overwrite the survey

Outcome: The survey's title is set to an empty string and its questions are set to an empty array.

### PR DESCRIPTION
**Pull request by MichaelDysart**
Name is now a primary key of a survey.
Close #30 
Close #46

### CODE REVIEW

File: src/test/java/webApp/webControllerTest.java [hunk 1]
```diff
@@ -21,11 +22,26 @@
...
 
         this.mockMvc.perform(post("/createSurvey").contentType("application/json")
                 .content(surveyString)).andExpect(status().isOk())
                 .andExpect(content().string(containsString("ok")))
                 .andExpect(content().string(containsString("done")));
+
+        this.mockMvc.perform(post("/createSurvey").contentType("application/json")
+                .content(surveyString)).andExpect(status().isOk())
+                .andExpect(content().string(containsString("error")))
+                .andExpect(content().string(containsString("Survey already exists with that name")));
```
Comment (TaylorAbraham) [inline src/test/java/webApp/webControllerTest.java]: Good job adding a test to ensure this bug does no reappear.

**Review by TaylorAbraham — COMMENTED** [review]
I understand that we are using name as a primary key given our way of searching for surveys is by name. But this seems like a poor limitation, restricting users to making unique survey names is not something that's ideal. Can we generate something like a very short UUID and use that to search instead? Because we can then use that UUID in the URL when we enable the functionality to generate links to surveys.

**Comment by TaylorAbraham** [comment]
So does this new update where you changed it back to using ID still fix the bug in question?

**Comment by MichaelDysart** [comment]
@TaylorAbraham yes, it does still fix the issue.
Tests in subsequent PRs do an even better job of ensuring valid input, including checking for null surveys, null questions, null question arrays, etc.

File: src/main/java/webApp/webController.java [hunk 2]
```diff
@@ -25,32 +27,42 @@ public String index(Model model){
...
         for (QuestionMessage question : surveyMessage.getQuestions()) {
+            questionTexts.add(question.getQuestion());
             if (question.getType().equals("openEnded")) {
                 questionList.add(new OpenEndedQuestion(question.getQuestion()));
             } else if(question.getType().equals("numberQuestion")) {
                 questionList.add(new NumberQuestion(question.getQuestion(), question.getMin(), question.getMax()));
             }
         }
 
+        HashSet<String> set = new HashSet<>(questionTexts);
```
Comment (JacksonSchoe) [inline src/main/java/webApp/webController.java:44]: I am happy to see this because I wanted a HashSet to be used for this information

**Review by JacksonSchoe — APPROVED** [review]
Could maybe use more inline comments, but honestly not that big of a deal. Overall good work.

**Review by rinadina — APPROVED** [review]
Looks fine, the addition of tests to the testRestApplication method appreciated

**Pull request merged**